src/main/java/com/atlassian/bitbucket/jenkins/internal/annotations/NotUpgradeSensitive.java

```diff
@@ -0,0 +1,18 @@
+package com.atlassian.bitbucket.jenkins.internal.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+
+/**
+ * Indicates that the class in question is not subject to upgrade concerns. This means it is not part of a job in any
+ * shape or form. It is not persisted in any way. Examples include Actions that displays a help link.
+ * @since 3.0.0
+ */
+@Target({TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface NotUpgradeSensitive {
```

Comment:
I noticed this isn't checked in the `testAnnotationsAreAllStillRelevant`. Do we intend for this annotation to persist, even after the plugin is released?

Reply:
Yes, I didn't really add many in this PR, but things like the `OAuthConsumerCreateAction` is not upgrade sensitive as it is not persisted. So these are marked "once and for all". We exclude them in the initial scan, but if they are included in a persisted class they are still checked.